### PR TITLE
Fix unread messages separator disapearing when marking channel as read

### PR DESCRIPTION
**Pull request by polqf**
### 🔗 Issue Links

https://github.com/GetStream/ios-issues-tracking/issues/564

### 🎯 Goal

- Fix regressions introduced by https://github.com/GetStream/stream-chat-swift/pull/2592
- Add a way to disable unread messages separator
- Make sure the channel is not marked as read until all unread messages are read

### 📝 Summary

When entering the chat, we want to keep the unread messages banner in the same place throughout the session, independently of whether the channel is marked as read.
It is only changed when marking as unread.

### 🛠 Implementation

UI logic has been split so jump to unread and mark as read (unread banner) are not tied completely

### 🧪 Scenarios

**Unread message in the last message in history**
- Should NOT mark as read upon opening
- Should display jump to unread pill
         - Should be displayed until reaching the first unread message
- Should display unread messages banner when scrolling to it
- Should mark as read when reaching the first unread message

**Unread message not in the first page**
- Should NOT mark as read upon opening
- Should display jump to unread pill
        - Should be displayed until reaching the first unread message
- Should display unread messages banner when scrolling to it
- Should mark as read when reaching the first unread message

**Unread message in the first page, but not visible**
- Should NOT mark as read upon opening
- Should display jump to unread pill
        - Should be displayed until reaching the first unread message
- Should display unread messages banner when scrolling to it
- Should mark as read when reaching the first unread message

**Unread message in the first page, that is visible**
- Should mark as read upon opening
- Should NOT display jump to unread pill
    - Should be displayed when the message is not on screen
- Should display unread messages banner in the right place
    - Unread message banner should stay during the whole session

**Mark as unread a message**
- Should NOT mark the channel as read in the current session
        - Not even when we see scroll to the first unread and back to bottom
- Should NOT display jump to unread pill
        - Should be displayed when the message is not on screen
- Should display unread messages banner in the right place
        - Unread message banner should stay during the whole session

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_

### CODE REVIEW

**Comment by nuno-vieira** [comment]
@polqf 
It seems like when there are multiple messages, and the current user receives new messages, they are not marked as read, and the unread top badge keeps showing. If we leave the channel, we can see the channel list says we have unread messages, although we were reading them while we received them:

https://github.com/GetStream/stream-chat-swift/assets/12814114/82578db6-3972-4012-8bdf-aaccd1dcf2e8

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,13 +4,17 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 # Upcoming
 
 ## StreamChatUI
+### ✅ Added
+- Unread messages separator can be enabled/disabled using `components.isUnreadMessagesSeparatorEnabled` (https://github.com/GetStream/stream-chat-swift/pull/2797)
```
Comment (nuno-vieira) [inline CHANGELOG.md:8]: This is going to be a hotfix, so having an addition here is a bit weird 🤔 But I guess this one is a minor one and useful

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Review by testableapple — APPROVED** [review]
LGTM 👍

**Pull request merged**